lib/srv/desktop/rdp/rdpclient/src/rdpdr/scard.rs

```diff
@@ -55,121 +59,227 @@ impl Client {
     }
 
     // ioctl handles messages coming from the RDP server over the RDPDR channel.
+    // TODO(isaiah): do we ever return None here? If not, we should change the return type.
     pub(super) fn ioctl(
         &mut self,
-        code: IoctlCode,
+        ioctl: &DeviceControlRequest,
         input: &mut Payload,
-    ) -> RdpResult<Option<Box<dyn Encode>>> {
-        debug!("got IoctlCode {:?}", &code);
+    ) -> RdpResult<Option<Vec<DeviceControlResponse>>> {
```

Comment:
Do we need `Option<Vec<` or can we just consider an empty vec to be the none case?

Same question applies for all of the occurrences below

Reply:
9d163707f73de19c3d05ffcb5a77636bd62268ce